### PR TITLE
EVAKA-4214: citizen sees message attachments

### PR DESCRIPTION
**Pull request by Joosakur**

### CODE REVIEW

File: frontend/src/lib-components/molecules/FileDownloadButton.tsx [hunk 1]
```diff
@@ -21,10 +25,23 @@ const DownloadButton = styled.button`
   text-decoration: none;
 `
 
+function getIcon(contentType: string): IconDefinition {
```
Comment (kimifoud) [inline frontend/src/lib-components/molecules/FileDownloadButton.tsx:28]: `FileUpload.tsx` exports `fileIcon` function in `lib-components/molecules`

Comment (Joosakur) [reply frontend/src/lib-components/molecules/FileDownloadButton.tsx:28]: oh, thanks 👍

**Comment by Joosakur** [comment]
e2e test passes locally but fails in ci. need to investigate..

File: service/src/main/resources/db/migration/R__message_account_access_view.sql [hunk 2]
```diff
@@ -1,22 +1,29 @@
 DROP VIEW IF EXISTS message_account_access_view;
 
-CREATE VIEW message_account_access_view(employee_id, account_id) AS (
-    SELECT employee.id AS employee_id, acc.id AS account_id
+CREATE VIEW message_account_access_view(account_id, employee_id, person_id) AS (
```
Comment (kimifoud) [inline service/src/main/resources/db/migration/R__message_account_access_view.sql:3]: I'd rather leave this view as it is (or maybe rename it to `employee_account_access_view`) and just check for person access directly in the queries. Or maybe create a separate view for citizen account access. There's no use-case where we lookup with both ids at the same time and this implementation has performance implications.

File: service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageQueries.kt [hunk 3]
```diff
@@ -220,9 +225,17 @@ SELECT
...
     msg.recipient_id,
     msg.recipient_name,
-    msg.recipient_account_type
+    msg.recipient_account_type,
+    (
+        SELECT coalesce(jsonb_agg(json_build_object(
+           'id', att.id,
+           'name', att.name,
+           'contentType', att.content_type
+        )) FILTER ( WHERE att.id IS NOT NULL ), '[]')
```
Comment (kimifoud) [inline service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageQueries.kt:234]: What is the `FILTER` clause needed for? 🤔

**Review by kimifoud — CHANGES_REQUESTED** [review]

File: service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageQueries.kt (hunk 3 above)
Comment (Joosakur) [reply service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageQueries.kt:234]: I think it otherwise produced an object with null values

Comment (kimifoud) [reply service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageQueries.kt:234]: Ach, the coalesce default should probably be `'[]'::jsonb`.

Comment (Joosakur) [reply service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageQueries.kt:234]: huh, that works

File: service/src/main/resources/db/migration/R__message_account_access_view.sql [hunk 4]
```diff
@@ -20,4 +20,4 @@ CREATE VIEW message_account_access_view(employee_id, account_id) AS (
     FROM message_account acc
         JOIN daycare_group_acl gacl ON gacl.daycare_group_id = acc.daycare_group_id
     WHERE acc.active = TRUE
-);
+);
```
Comment (kimifoud) [inline service/src/main/resources/db/migration/R__message_account_access_view.sql:23]: 🤓 
```suggestion
);

```

**Review by kimifoud — APPROVED** [review]

**Pull request merged**